beacon-chain/sync/subscriber_beacon_blocks.go

```diff
@@ -108,6 +109,23 @@ func (s *Service) processDataColumnSidecarsFromExecution(ctx context.Context, ro
 		log.Warning("Data column storage is not enabled, skip saving data column, but continue to reconstruct and broadcast data column")
 	}
 
+	// Check if data is already available to avoid unnecessary execution client calls
+	roBlock, err := blocks.NewROBlockWithRoot(roSignedBlock, blockRoot)
+	if err != nil {
+		log.WithError(err).Error("Failed to create ROBlock")
+		return
+	}
+	switch err := s.cfg.chain.IsDataAvailable(ctx, roBlock); {
+	case err == nil:
+		log.Debug("Data already available – skipping execution-client call")
+		return
+	case errors.Is(err, blockchain.ErrDataNotAvailable):
+		// continue
+	default:
+		log.WithError(err).Error("Failed to check data availability")
+		return
+	}
+
 	// When this function is called, it's from the time when the block is received, so in almost all situations we need to get the data column from EL instead of the blob storage.
 	sidecars, err := s.cfg.executionReconstructor.ReconstructDataColumnSidecars(ctx, roSignedBlock, blockRoot)
```

Comment:
Is it possible to use `roBlock` (which has a `.Root()` cheap method) instead of `roSignedBlock` and `blockRoot`?